env/fs_posix.cc

```diff
@@ -1043,6 +1043,87 @@ class PosixFileSystem : public FileSystem {
   }
 #endif  // ROCKSDB_IOURING_PRESENT
 
+  virtual IOStatus Poll(std::vector<void*>& io_handles,
+                        size_t min_completions) override {
+    IOStatus io_s;
+#if defined(ROCKSDB_IOURING_PRESENT)
+    // io_uring_queue_init.
+    struct io_uring* iu = nullptr;
+    if (thread_local_io_urings_) {
+      iu = static_cast<struct io_uring*>(thread_local_io_urings_->Get());
+      if (iu == nullptr) {
+        iu = CreateIOUring();
+        if (iu != nullptr) {
+          thread_local_io_urings_->Reset(iu);
+        }
+      }
+    }
+
+    // Init failed, platform doesn't support io_uring.
+    if (iu == nullptr) {
+      return IOStatus::NotSupported("Poll");
+    }
+
+    size_t count = 0;
+    for (size_t i = 0; i < io_handles.size() && count < min_completions; i++) {
+      // The request has been completed in earlier runs.
+      if ((static_cast<Posix_IOHandle*>(io_handles[i]))->is_finished) {
+        count++;
+        continue;
+      }
+      // Loop until IO for io_handles[i] is completed.
+      while (true) {
+        // io_uring_wait_cqe.
+        struct io_uring_cqe* cqe = nullptr;
+        ssize_t ret = io_uring_wait_cqe(iu, &cqe);
+        if (ret) {
+          io_s = IOStatus::IOError("io_uring_wait_cqe() returns " +
```

Comment:
I'm not sure if there's a good way to handle such an error. It might be safer to call abort(), rather than get into some indeterminate state.